awx/main/migrations/0113_v370_event_bigint.py

```diff
@@ -22,11 +22,8 @@ def migrate_event_data(apps, schema_editor):
             # recreate counter for the new table's primary key to
             # start where the *old* table left off (we have to do this because the
             # counter changed from an int to a bigint)
-            cursor.execute(f'DROP SEQUENCE IF EXISTS "{tblname}_id_seq" CASCADE;')
-            cursor.execute(f'CREATE SEQUENCE "{tblname}_id_seq";')
-            cursor.execute(f'ALTER TABLE "{tblname}" ALTER COLUMN "id" ' f"SET DEFAULT nextval('{tblname}_id_seq');")
-            cursor.execute(f"SELECT setval('{tblname}_id_seq', (SELECT MAX(id) FROM _old_{tblname}), true);")
-            cursor.execute(f'DROP TABLE _old_{tblname};')
+            cursor.execute(f'CREATE SEQUENCE IF NOT EXISTS "{tblname}_id_seq";')
+            cursor.execute(f"SELECT setval('{tblname}_id_seq', COALESCE((SELECT MAX(id)+1 FROM _old_{tblname}), 1), false);")
```

Comment:
Thanks for calling this out. I'm actually going to revert this file as it was more of a problem with the psycopg3 upgrade which is no longer a thing in this version of the PR.

Reply:
I retract my previous statement about this being psycopg3. This was from the Django upgrade. When you try to run this with original lines you get this stack:
```
tools_awx_1 |   Applying main.0113_v370_event_bigint...Traceback (most recent call last):
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
tools_awx_1 |     return self.cursor.execute(sql)
tools_awx_1 | psycopg2.errors.DependentObjectsStillExist: cannot drop sequence main_jobevent_id_seq because column id of table _old_main_jobevent requires it
tools_awx_1 | HINT:  You can drop column id of table _old_main_jobevent instead.
tools_awx_1 | 
tools_awx_1 | 
tools_awx_1 | The above exception was the direct cause of the following exception:
tools_awx_1 | 
tools_awx_1 | Traceback (most recent call last):
tools_awx_1 |   File "/usr/local/bin/awx-manage", line 9, in <module>
tools_awx_1 |     load_entry_point('awx', 'console_scripts', 'awx-manage')()
tools_awx_1 |   File "/awx_devel/awx/__init__.py", line 200, in manage
tools_awx_1 |     execute_from_command_line(sys.argv)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/__init__.py", line 442, in execute_from_command_line
tools_awx_1 |     utility.execute()
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/__init__.py", line 436, in execute
tools_awx_1 |     self.fetch_command(subcommand).run_from_argv(self.argv)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/base.py", line 412, in run_from_argv
tools_awx_1 |     self.execute(*args, **cmd_options)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/base.py", line 458, in execute
tools_awx_1 |     output = self.handle(*args, **options)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/base.py", line 106, in wrapper
tools_awx_1 |     res = handle_func(*args, **kwargs)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/core/management/commands/migrate.py", line 356, in handle
tools_awx_1 |     post_migrate_state = executor.migrate(
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/migrations/executor.py", line 135, in migrate
tools_awx_1 |     state = self._migrate_all_forwards(
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/migrations/executor.py", line 167, in _migrate_all_forwards
tools_awx_1 |     state = self.apply_migration(
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/migrations/executor.py", line 252, in apply_migration
tools_awx_1 |     state = migration.apply(state, schema_editor)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/migrations/migration.py", line 132, in apply
tools_awx_1 |     operation.database_forwards(
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/migrations/operations/special.py", line 193, in database_forwards
tools_awx_1 |     self.code(from_state.apps, schema_editor)
tools_awx_1 |   File "/awx_devel/awx/main/migrations/0113_v370_event_bigint.py", line 25, in migrate_event_data
tools_awx_1 |     cursor.execute(f'DROP SEQUENCE IF EXISTS "{tblname}_id_seq" CASCADE;')
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 102, in execute
tools_awx_1 |     return super().execute(sql, params)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 67, in execute
tools_awx_1 |     return self._execute_with_wrappers(
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 80, in _execute_with_wrappers
tools_awx_1 |     return executor(sql, params, many, context)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 89, in _execute
tools_awx_1 |     return self.cursor.execute(sql, params)
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/utils.py", line 91, in __exit__
tools_awx_1 |     raise dj_exc_value.with_traceback(traceback) from exc_value
tools_awx_1 |   File "/var/lib/awx/venv/awx/lib64/python3.9/site-packages/django/db/backends/utils.py", line 87, in _execute
tools_awx_1 |     return self.cursor.execute(sql)
tools_awx_1 | django.db.utils.InternalError: cannot drop sequence main_jobevent_id_seq because column id of table _old_main_jobevent requires it
tools_awx_1 | HINT:  You can drop column id of table _old_main_jobevent instead.
```

In `devel` we see these are only NOTICE/DETAILS and not an ERROR:
```
tools_postgres_1 | 2023-05-11 15:14:28.033 UTC [86] NOTICE:  drop cascades to 2 other objects
tools_postgres_1 | 2023-05-11 15:14:28.033 UTC [86] DETAIL:  drop cascades to default value for column id of table _old_main_jobevent
tools_postgres_1 | 	drop cascades to default value for column id of table main_jobevent
tools_postgres_1 | 2023-05-11 15:14:28.072 UTC [86] NOTICE:  drop cascades to 2 other objects
tools_postgres_1 | 2023-05-11 15:14:28.072 UTC [86] DETAIL:  drop cascades to default value for column id of table _old_main_inventoryupdateevent
tools_postgres_1 | 	drop cascades to default value for column id of table main_inventoryupdateevent
tools_postgres_1 | 2023-05-11 15:14:28.125 UTC [86] NOTICE:  drop cascades to 2 other objects
tools_postgres_1 | 2023-05-11 15:14:28.125 UTC [86] DETAIL:  drop cascades to default value for column id of table _old_main_projectupdateevent
tools_postgres_1 | 	drop cascades to default value for column id of table main_projectupdateevent
tools_postgres_1 | 2023-05-11 15:14:28.190 UTC [86] NOTICE:  drop cascades to 2 other objects
tools_postgres_1 | 2023-05-11 15:14:28.190 UTC [86] DETAIL:  drop cascades to default value for column id of table _old_main_adhoccommandevent
tools_postgres_1 | 	drop cascades to default value for column id of table main_adhoccommandevent
tools_postgres_1 | 2023-05-11 15:14:28.238 UTC [86] NOTICE:  drop cascades to 2 other objects
tools_postgres_1 | 2023-05-11 15:14:28.238 UTC [86] DETAIL:  drop cascades to default value for column id of table _old_main_systemjobevent
tools_postgres_1 | 	drop cascades to default value for column id of table main_systemjobevent
```

Its unclear why the Django upgrade changes this from Notice to Error but this is why we had to alter these statements. 

However, since no one should be migrating from a version this old directly to this new version we can simplify this area greatly by just doing an alter table. Here is some details from psql showing that we can alter an int id to a bigint id and the sequence will also update:
```
awx=# CREATE TABLE test_table (id int GENERATED ALWAYS AS IDENTITY, value varchar);
CREATE TABLE
awx=# \d test_table_id_seq
                 Sequence "public.test_table_id_seq"
  Type   | Start | Minimum |  Maximum   | Increment | Cycles? | Cache 
---------+-------+---------+------------+-----------+---------+-------
 integer |     1 |       1 | 2147483647 |         1 | no      |     1
Sequence for identity column: public.test_table.id
awx=# ALTER TABLE test_table ALTER COLUMN id TYPE bigint USING id::bigint;
ALTER TABLE
awx=# \d test_table_id_seq
                     Sequence "public.test_table_id_seq"
  Type  | Start | Minimum |       Maximum       | Increment | Cycles? | Cache 
--------+-------+---------+---------------------+-----------+---------+-------
 bigint |     1 |       1 | 9223372036854775807 |         1 | no      |     1
Sequence for identity column: public.test_table.id
```

Basically just altering the tables ID column to a bigint changes the sequence for us. Technically I think this would work for an upgrade scenario as well but since we truly only care about new table creation we can greatly simplify this to get around the error produced by Django 4.2/postgres.